src/gtc/common.py

```diff
@@ -572,6 +572,46 @@ def apply(cls, node: Node, *, symtable: Dict[str, Any]) -> List[str]:
     return root_validator(allow_reuse=True, skip_on_failure=True)(_impl)
 
 
+def validate_lvalue_dims() -> RootValidatorType:
+    """Validate lvalue dimensions using the root node symbol table."""
+
+    class _LvalueDimsValidator(NodeVisitor):
+        def visit_Node(self, node: Node, symtable: Dict[str, Any], **kwargs: Any) -> None:
+            if hasattr(node, "loop_order"):
+                kwargs["loop_order"] = node.loop_order
+            if isinstance(node, SymbolTableTrait):
+                symtable = {**symtable, **node.symtable_}
+            self.generic_visit(node, symtable=symtable, **kwargs)
+
+        def visit_AssignStmt(
+            self, node: AssignStmt, *, loop_order: LoopOrder, **kwargs: Any
+        ) -> None:
+            symtable = kwargs["symtable"]
+            allowed_masks = [(True, True, True)]  # ijk always allowed
+            if loop_order is not LoopOrder.PARALLEL:
+                allowed_masks.append((True, True, False))  # ij only allowed in FORWARD and BACKWARD
+            name = node.left.name
+            if name in symtable and hasattr(symtable[name], "dimensions"):
```

Comment:
Shouldn't `left` always have a `dimensions` field (with default `(True, True, True)`), removing the need for this?

Reply:
`left` is usually a `FieldAccess` instance, which doesn't have a `dimensions` field. Basically the dimensions are set when the field is declared and since it can never change after that there is no point duplicating that info on every reference to it.